### DISCUSSION TITLE
Use vcpkg in a Visual Studio project without user-wide integration

### DISCUSSION THREAD

**Discussion by T-Dimov** [microsoft/vcpkg#39270] [Q&A]
We don't have enough information to solve this issue, unfortunately; if you have any more information to help us solve this, please reopen!

_Originally posted by @JackBoosY in https://github.com/microsoft/vcpkg/issues/18868#issuecomment-930830469_
            
I think I have the same problem. Our software uses libcurl by just copying its headers and library files into our source tree and referencing said copies. For a side project I installed cpr trough vcpkg which also brought along libcurl. If I integrate vcpkg user-wide the main software fails to link because now it's able to find 2 copies of libcurl and can't decide which one to link. Modifying it is not an option. I'd like to integrate vcpkg for a single solution/project.

**Comment by StarGate-One** [comment]
See if `vcpkg integrate project` can do what you need.

> ->vcpkg help integrate
>Synopsis: Integrates vcpkg with machines, projects, or shells
>Examples:
  vcpkg integrate install
                         Makes installed packages available user-wide. Requires admin privileges on
                         first use
  vcpkg integrate remove Removes user-wide integration
>
>  _vcpkg integrate project
                         Generates a referencing NuGet package for individual Visual Studio project
                         use. Windows-only_

**Comment by T-Dimov** [comment]
According to the documentation NuGet isn't recommended. Admittedly, it is going to be Windows-only anyway but I think there should be an easy cross-platform way to integrate into a project just like `vcpkg integrate install` does user-wide.

**Reply by AshwinKumarVijay** [reply]
Yeah, I think this is something that should be addressed by the vcpkg team - it makes a lot of sense, and it would make it a lot easier for repositories to depend on vcpkg without running the risk of affecting other repositories that depend on vcpkg. Just look at how much use venvs get in python package management.
